util.c

```diff
@@ -2311,7 +2313,9 @@ static bool parse_notify(struct pool *pool, json_t *val)
 	pool->weight = weight;
 	pool->nonce_size = nonce_size;
 
-	pool->nonce2 = 0;
+	if (clean) {
+		pool->nonce2 = 0;
+	}
```

Comment:
This seems to be the only side-effect of this "clean" param. But I'd expect it to have the same effect as the original stratum clean param, which if I understand correctly is a bit different.

The original "clean" param. Would make cgminer drop the current share in favor of mining shares for the newly received job, before waiting for the current share to finish. Otherwise the newly received job would start after the current share is finished. But I have no idea how this is implemented.

If this isn't supposed to do this, I'd say we're better off using a different param name. Additionally a simple explanation somewhere (PR description, commit message, code comment) would be really useful.

Reply:
I think you missed the `pool->swork.clean = clean;` line above. I haven't checked exactly what it does because I just followed the code in the original cgminer source code, but I can do it.

Reply:
The original cgminer code: https://github.com/vthoang/cgminer/blob/master/util.c#L2297-L2325 seems to just set `pool->swork.clean = clean;` and the rest of the code is prepared to use it. Maybe it would be enough for our case.

Reply:
I based my changes on our fork repository, just before our first commit was merged.

Reply:
Here is the link: https://github.com/HathorNetwork/cgminer/commit/7810bb22944c7b00291e0559f9feb80e57fa4d51#diff-6d7e72ea1ca2b7d0258e568a155084bdL2324

Reply:
I just noticed that these changes restore the original behavior. I'm OK with this then.